solutions/usaco-898.mdx

```diff
@@ -12,7 +12,85 @@ author: Nathan Gong, Melody Yu (Video)
 Note: The video solution might not be the same as other solutions. Code in C++.
 <Youtube id="6oyl7lTKdPY" />
 
-## Implementation
+## Implementation (Indexed Set)
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+First, read the official analysis. The editorial intends for a data structure that
+
+> can count the number of elements of $S$ which are smaller than $y$.
+
+Rather than using a segment tree, we can instead use `order_of_key` to perform such queries in $\mathcal{O}(\log N)$ time.
```

Comment:
```suggestion
Rather than using a segment tree, we can instead use an order statistic tree to perform such queries in $\mathcal{O}(\log N)$ time.
```